custom/icds_reports/utils/aggregation_helpers/distributed/agg_bihar_api_demographics.py

```diff
@@ -0,0 +1,110 @@
+            ('ifsc_code', 'person_list.ifsc_code'),
+            ('age_at_reg', 'person_list.age_at_reg'),
+            ('dob', 'person_list.dob'),
+            ('gender', 'person_list.sex'),
+            ('blood_group', 'person_list.blood_group'),
+            ('disabled', 'person_list.disabled'),
+            ('disability_type', 'person_list.disability_type'),
+            ('referral_status', 'person_list.referral_status'),
+            ('migration_status', 'migration_tab.is_migrated'),
+            ('resident', 'person_list.resident'),
+            ('registered_status', 'person_list.registered_status'),
+            ('rch_id', 'person_list.rch_id'),
+            ('mcts_id', 'person_list.mcts_id'),
+            ('phone_number', 'person_list.phone_number'),
+            ('date_death', 'person_list.date_death'),
+            ('site_death', 'person_list.site_death'),
+            ('closed_on', 'person_list.closed_on'),
+            ('reason_closure', 'person_list.reason_closure'),
+            ('household_id', 'hh_list.doc_id'),
+            ('household_name', 'hh_list.name'),
+            ('hh_reg_date', 'hh_list.hh_reg_date'),
+            ('hh_num', 'hh_list.hh_num'),
+            ('hh_gps_location', 'hh_list.hh_gps_location'),
+            ('hh_caste', 'hh_list.hh_caste'),
+            ('hh_bpl_apl', 'hh_list.hh_bpl_apl'),
+            ('hh_minority', 'hh_list.hh_minority'),
+            ('hh_religion', 'hh_list.hh_religion'),
+
+        )
+        column_names = ", ".join([col[0] for col in columns])
+        calculations = ", ".join([col[1] for col in columns])
+
+        return f"""
+                INSERT INTO "{self.tablename}" (
+                    {column_names}
+                )
+                (
+                SELECT
+                {calculations}
+                from "{person_case_ucr}" person_list
+                LEFT JOIN "{AGG_MIGRATION_TABLE}" migration_tab ON (
+                    person_list.doc_id = migration_tab.person_case_id AND
+                    person_list.supervisor_id = migration_tab.supervisor_id AND
+                    migration_tab.month='{month_start_string}'
+                )
+                LEFT JOIN "{household_ucr}" hh_list ON (
+                    person_list.household_case_id = hh_list.doc_id AND
+                    person_list.supervisor_id = hh_list.supervisor_id
+                )
+                WHERE (person_list.opened_on <= '{month_end_string}' AND
+              (person_list.closed_on IS NULL OR person_list.closed_on >= '{month_start_string}' )) AND
+              (
+                migration_tab.is_migrated is distinct from 1 OR
+                migration_tab.migration_date>='{month_start_string}')
+              );
+                """
+
+    def indexes(self):
+        return [
+            'CREATE INDEX IF NOT EXISTS demographics_state_person_case_idx ON "{}" (month, state_id, person_id)'.format(self.tablename)
```

Comment:
This index because most of the query which will be happening on this will be like `where month=x and state=y and order by person_id`

Reply:
hey, do you think I need to add supervisor_id as well in the index because I will be joining the table with awc_location table on matching supervisor_id as well to make the query distributed?
@calellowitz

Reply:
It is generally a good idea to have an index on columns used in the join. It is hard to say specifically in this instance which is more important, and I don't know if both can be used in the query. It may be worth testing out a bit to see which is better once the table is made.